test/with_dummyserver/test_socketlevel.py

```diff
@@ -1427,6 +1427,147 @@ def test_load_verify_locations_exception(self):
         with pytest.raises(SSLError):
             ssl_wrap_socket(None, ca_certs="/tmp/fake-file")
 
+    from contextlib import contextmanager
+
+    @contextmanager
+    def _simple_https_pool(self, *args, **kwargs):
+        with HTTPSConnectionPool(
+            self.host, self.port, *args, retries=False, **kwargs
+        ) as pool:
+            yield pool
+
+    def test_load_verify_locations_cadata(self, good_ca_chain):
+        """
+        Ensure that load_verify_locations can load cadata for all backends
+        """
+        ca_cert, server_cert, server_key = good_ca_chain
+
+        def socket_handler(listener):
+            for i in range(2):
+                sock = listener.accept()[0]
+                try:
+                    ssl_sock = ssl.wrap_socket(
+                        sock,
+                        server_side=True,
+                        keyfile=server_key["file"],
+                        certfile=server_cert["file"],
+                        ca_certs=ca_cert["file"],
+                    )
+                except Exception:
+                    sock.close()
+                    continue
+
+                buf = b""
+                while not buf.endswith(b"\r\n\r\n"):
+                    buf += ssl_sock.recv(65536)
+
+                ssl_sock.send(
+                    b"HTTP/1.1 200 OK\r\n"
+                    b"Content-Type: text/plain\r\n"
+                    b"Content-Length: 0\r\n\r\n"
+                )
+
+                ssl_sock.close()
+
+        self._start_server(socket_handler)
+
+        # fail with wrong CA
+        with self._simple_https_pool(ca_certs=DEFAULT_CA) as pool:
+            with pytest.raises(SSLError):
+                pool.request("GET", "/", timeout=LONG_TIMEOUT)
+
+        # data
+        with self._simple_https_pool(ca_cert_data=ca_cert["data"]) as pool:
+            r = pool.request("GET", "/", timeout=LONG_TIMEOUT)
+            assert r.status == 200
+
+    @pytest.mark.parametrize(
+        "cafile,capath,cadata",
```

Comment:
Instead of all permutations, what are your thoughts on stacking 3 `@parametrize("cafile", [True, False, None]`?